### PR TITLE
(PUP-5545) rubocop by default

### PR DESCRIPTION
**Pull request by ghoneycutt**

### CODE REVIEW

**Comment by HAIL9000** [comment]
Hey @ghoneycutt, it looks like the Rubocop check is now failing in Travis, could you take a quick look and see what's going on there?

**Comment by ghoneycutt** [comment]
Hi @HAIL9000 

Seems that the rubocop testing for puppet itself is using an older version and was attempting to read the .rubocop.yml file in the skeleton. I added a commit which tells the rubocop for the puppet project to ignore files in the puppet skeleton itself.

https://github.com/puppetlabs/puppet/pull/4463/files#diff-11a0d7906801d9dea0eccb85667ad811R15

**Comment by ghoneycutt** [comment]
Don't merge yet.. I found some more additions to the configuration while testing another module that contained custom facts. I'll also need to run this over a module with custom types and providers to ensure it isn't suggesting changes for things that are common place in the community.

**Comment by kylog** [comment]
Ping @andersonmills

File: lib/puppet/module_tool/skeleton/templates/generator/.rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
```
Comment (adreyer) [inline lib/puppet/module_tool/skeleton/templates/generator/.rubocop.yml]: @hunner Is this the style y'all use?

**Comment by MikaelSmith** [comment]
Needs a rebase :(

**Comment by HAIL9000** [comment]
Hey @ghoneycutt, do you have some time to give this a quick rebase? Looks like there are merge conflicts

**Comment by joshcooper** [comment]
Hey @ghoneycutt could you rebase? @andersonmills can you follow up and either merge when the changes become available or close if there is no activity?

**Comment by ghoneycutt** [comment]
@joshcooper @hail9000

Rebased

File: lib/puppet/module_tool/skeleton/templates/generator/.rubocop.yml (hunk 1 above)
Comment (hunner) [inline lib/puppet/module_tool/skeleton/templates/generator/.rubocop.yml]: I don't think we have a specified format yet, but voxpupuli has prior-art on this and it appears to be the same https://github.com/voxpupuli/modulesync_config/blob/master/moduleroot/.rubocop.yml 👍

**Comment by MikaelSmith** [comment]
I've opened a rebased PR at https://github.com/puppetlabs/puppet/pull/4936 for @andersonmills to test. If it looks good we'll merge it, otherwise might have more requests for work.

**Comment by andersonmills** [comment]
Closed in deference to #4936

**Pull request closed without merging**